File: лр2/код.java
// Создание мультисписка (вложенного списка):
public class Node {
    int data;
    Node prev;
    Node next;

    public Node(int data)
    {
        this.data = data;
        this.prev = null;
        this.next = null;
    }
}

public class DoublyLinkedList {
    Node head;
    Node tail;

    public DoublyLinkedList()
    {
        this.head = null;
        this.tail = null;
    }
}


// Создание очереди:
Queue<String> queue = new LinkedList<>();
queue.add("клубника"); 
queue.add("апельсин"); 
queue.add("арбуз"); 


//Создание дека:
Deque<Integer> stack = new ArrayDeque<>();
stack.push(1); 
stack.push(2); 
stack.push(3); 


//Создание приоритетной очереди:
PriorityQueue<Integer> minHeap = new PriorityQueue<>();
minHeap.offer(20); 
minHeap.offer(15); 
minHeap.offer(25); 
minHeap.offer(4); 


//Создание приоритетной очереди с компаратором:
PriorityQueue<Task> priorityQueue = new PriorityQueue<>(idComparator);
priorityQueue.add(new Task(10001, "Task 1", 15)); 
priorityQueue.add(new Task(10003, "Task 3", 20)); 
priorityQueue.add(new Task(10002, "Task 2", 2)); 

File: лр2/код.cpp
// Создание мультисписка (вложенного списка):
struct Node {
    int data;
    Node *prev;
    Node *next;
};

// Создание очереди:
std::queue<std::string> queue;
queue.push("Ulya"); 
queue.push("Alya"); 
queue.push("Sonya");

//Создание дека:
std::deque<int> d = {1, 2, 3};

//Создание приоритетной очереди с примитивным типом:
std::priority_queue<int> pq;
pq.push(20);     
pq.push(40);    
pq.push(5); 

//Создание приоритетной очереди с пользовательскими структурами:
std::priority_queue<Task> taskQueue;
taskQueue.push({"Task 1", 2});     
taskQueue.push({"Task 2", 1});     
taskQueue.push({"Task 3", 3});  
